pkg/healthcheck/healthcheck.go

```diff
@@ -827,47 +826,19 @@ func (hc *HealthChecker) validateServiceProfiles() error {
 	}
 
 	for _, p := range svcProfiles.Items {
-		nameParts := strings.Split(p.Name, ".")
-		if len(nameParts) != 2+len(clusterZoneSuffix) {
-			return fmt.Errorf("ServiceProfile \"%s\" has invalid name (must be \"<service>.<namespace>.svc.cluster.local\")", p.Name)
-		}
-		for i, part := range nameParts[2:] {
-			if part != clusterZoneSuffix[i] {
-				return fmt.Errorf("ServiceProfile \"%s\" has invalid name (must be \"<service>.<namespace>.svc.cluster.local\")", p.Name)
-			}
+		service, namespace, err := profiles.ValidateName(p.Name)
+		if err != nil {
+			return err
 		}
-		service := nameParts[0]
-		namespace := nameParts[1]
-		_, err := hc.clientset.Core().Services(namespace).Get(service, meta_v1.GetOptions{})
+
+		_, err = hc.clientset.Core().Services(namespace).Get(service, meta_v1.GetOptions{})
```

Comment:
We should be explicit about the version of `CoreClient` here

```suggestion
		_, err = hc.clientset.CoreV1().Services(namespace).Get(service, meta_v1.GetOptions{})
```